libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -181,7 +181,11 @@ bool AP_Mount_Siyi::healthy() const
 // get attitude as a quaternion.  returns true on success
 bool AP_Mount_Siyi::get_attitude_quaternion(Quaternion& att_quat)
 {
-    att_quat.from_euler(_current_angle_rad.x, _current_angle_rad.y, _current_angle_rad.z);
+    // Order of rotation for Siyi gimbals is (yaw, roll, pitch), which is 312 order
```

Comment:
OK, since PaulR's given his blessing as well, I think we can merge this.  A small style thing is I think we could skip the intermediate variables and just directly pass them into the from_vector312 function.

Reply:
Done.